Pyic 8332: Updated spinner component for DAD journey.

undermad opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Fixed link behaviour in detail components
Changed spinner component for DAD journey to align with Figma design. 
Fixed 'Continue' button on spinner page in MAM journey to be correctly translated to Welsh language.
Removed contact link from MAM spinner screens

### Why did it change

To correctly align website with design requirements.

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-8332](https://govukverify.atlassian.net/browse/PYIC-8332)

## Checklists

<!-- Delete if changes don't apply -->
- [x] Browser/unit tests have been written/updated


[PYIC-8332]: https://govukverify.atlassian.net/browse/PYIC-8332?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

undermad committed:
Fixed traige-desktop-download-app to not open new tab when link is clicked inside the componend, instead it will lunch it in the same tab.

undermad committed:
PYIC-8332: Changed spinner to consist of 2 states only and changed messeges for each state.

undermad committed:
PYIC-8332: Changed spinner component to consist of 2 states in DAD journey.

undermad committed:
PYIC-8332: Removed contact link from page that contains spinner component in MAM journey.

undermad committed:
PYIC-8332: Added text content for continue button in spinner componend for MAM journey

undermad committed:
PYIC-8332: Fixed missing syntax

undermad committed:
PYIC-8332: Aligned continue button attribute to match new spinner component version.

undermad committed:
PYIC-8332: Updated spinner version to 1.1.1

undermad committed:
PYIC-8332: Fixed translation to Welsh

undermad committed:
PYIC-8332: Updated snapshot tests.

undermad committed:
PYIC-8332: Changed links to open in same tab for Welsh website.

undermad committed:
PYIC-8332: Updated functional tests.

MikeCollingwood commented on browser-tests/functional-tests/functional.spec.ts:
```diff
@@ -201,12 +201,15 @@ test.describe.parallel("Functional tests", () => {
 
   test.describe.parallel("Check Strategic App VC receipt", () => {
     ["Mam", "Dad"].forEach((journeyType) => {
```
```suggestion
    [{
      journeyType: "Mam",
      inProgressSpinnerText: "This may take a few minutes.",
      completedSpinnerText: "You can now continue.",
    }, ...].forEach((journeyType) => {
```

So you can remove logic from within the test.

MikeCollingwood commented on browser-tests/functional-tests/functional.spec.ts:
```diff
@@ -271,12 +274,15 @@ test.describe.parallel("Functional tests", () => {
-        const longWaitSpinnerTextLocator = await page.getByText("We’re still checking your details. Do not close or refresh this page.");
-        await expect(longWaitSpinnerTextLocator).toBeVisible({ timeout: 65000 });
+        if(journeyType === "Mam") {
+          const longWaitSpinnerTextLocator = await page.getByText("We’re still checking your details. Do not close or refresh this page.");
```
A test specifically for checking the long-wait text should be separated and only for Mam

MikeCollingwood commented on locales/en/translation.json:
```diff
@@ -1165,7 +1174,7 @@
         "details4": {
           "summaryText": "I’d prefer not to use the app",
-          "html": "<p>If you’d prefer not to use the app, you may be able to <a class=\"govuk-link\" target=\"_blank\" rel=\"noopener noreferrer\" href=\"/ipv/journey/pyi-triage-mobile-download-app/preferNoApp\">prove your identity another way</a>.</p>"
+          "html": "<p>If you’d prefer not to use the app, you may be able to <a class=\"govuk-link\" lrel=\"noopener noreferrer\" href=\"/ipv/journey/pyi-triage-mobile-download-app/preferNoApp\">prove your identity another way</a>.</p>"
```
```suggestion
          "html": "<p>If you’d prefer not to use the app, you may be able to <a class=\"govuk-link\" rel=\"noopener noreferrer\" href=\"/ipv/journey/pyi-triage-mobile-download-app/preferNoApp\">prove your identity another way</a>.</p>"
```

undermad committed:
PYIC-8332: Improved code quality.

blakeyp commented on locales/cy/translation.json:
```diff
@@ -1125,21 +1128,30 @@
+        "paragraph6": "Yna gallwch barhau i wasanaeth y llywodraeth rydych chi am ei ddefnyddio.",
+        "spinner": {
+          "initial": {
+            "spinnerStateText": "Ar y gweill",
```
If this isn't the official translation can we keep it as the English "In progress" for now (perhaps with "TODO-" in front) - so we don't think it's already been done :)
These pages aren't live yet so it's fine for this to be merged and then a new PR raised to add the translation once we receive it

undermad committed:
PYIC-8332: Change spinner text to English version in Welsh translation.

We have to wait for Welsh translation for 'In progress' phrase. To complete the ticket in time we decided to change temporarily Welsh version to English as this content is not live yet.
